examples/contrib/forecast/bart.py

```diff
@@ -0,0 +1,149 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import logging
+
+import numpy as np
+import torch
+
+import pyro
+import pyro.distributions as dist
+from pyro.contrib.examples.bart import load_bart_od
+from pyro.contrib.forecast import ForecastingModel, backtest
+from pyro.ops.tensor_utils import periodic_cumsum, periodic_repeat
+
+logging.getLogger("pyro").setLevel(logging.DEBUG)
+logging.getLogger("pyro").handlers[0].setLevel(logging.DEBUG)
+
+
+def preprocess(args):
+    """
+    Extract a tensor of (arrivals,departures) to Embarcadero station.
+    """
+    print("Loading data")
+    dataset = load_bart_od()
+    i = dataset["stations"].index("EMBR")
+    arrivals = dataset["counts"][:, :, i].sum(-1)
+    departures = dataset["counts"][:, i, :].sum(-1)
+    data = torch.stack([arrivals, departures], dim=-1)
+    covariates = torch.zeros(len(data), 0)  # Has correct len() but no elements.
+    return data, covariates
+
+
+# We define a model by subclassing the ForecastingModel class and implementing
+# a single .model() method.
+class Model(ForecastingModel):
+    # The .model() method inputs two tensors: a fake tensor zero_data that is
+    # the same size and dtype as the real data (but of course the generative
+    # model shouldn't depend on the value of the data it generates!), and a
+    # tensor of covariates. Our simple model depends on no covariates, so we
+    # simply pass in an empty tensor (see  the preprocess() function above).
+    def model(self, zero_data, covariates):
+        duration = zero_data.size(-2)
+        period = 24 * 7
+
+        # Sample global parameters.
+        noise_scale = pyro.sample("noise_scale",
+                                  dist.LogNormal(torch.full((2,), -3), 1).to_event(1))
+        trans_timescale = pyro.sample("trans_timescale",
+                                      dist.LogNormal(torch.zeros(2), 1).to_event(1))
+
+        trans_scale = pyro.sample("trans_scale",
+                                  dist.LogNormal(torch.zeros(2), 0.1).to_event(1))
+        trans_corr = pyro.sample("trans_corr",
+                                 dist.LKJCorrCholesky(2, torch.ones(())))
+        trans_scale_tril = trans_scale.unsqueeze(-1) * trans_corr
+
+        obs_scale = pyro.sample("obs_scale",
+                                dist.LogNormal(torch.zeros(2), 0.1).to_event(1))
+        obs_corr = pyro.sample("obs_corr",
+                               dist.LKJCorrCholesky(2, torch.ones(())))
+        obs_scale_tril = obs_scale.unsqueeze(-1) * obs_corr
```

Comment:
@fehiepsi did I get this linear algebra correct, for sampling a `scale_tril` matrix?

Reply:
would it be helpful to pepper this with a bunch of assert shape statements? (it would be to me anyway)

Reply:
done.